### PR TITLE
Fix hivemind cooldown exploit

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Title. Closes https://github.com/tgstation/TerraGov-Marine-Corps/issues/8681

## Changelog
:cl:
fix: Fix hivemind cooldown exploit
/:cl:

### CODE REVIEW

**Comment by Mrrpip** [comment]
Oh well. It was my big brain counter to razorwire spam while it lasted!

*salute

**Comment by BraveMole** [comment]
> Oh well. It was my big brain counter to razorwire spam while it lasted!
> 
> *salute

probably gonna make a cooldown buff after

File: code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm [hunk 1]
```diff
@@ -138,11 +141,24 @@
...
 	set_datum(FALSE)
-	remove_abilities()
-	add_abilities()
+	swap_abilities()
 	update_wounds()
 	update_icon()
 
+///Give hivemind the correct abilities for its form
+/mob/living/carbon/xenomorph/hivemind/proc/swap_abilities()
+	var/list/datum/action/chached_actions = other_actions?.Copy()
```
Comment (psykzz) [inline code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm:150]:
```suggestion
	var/list/datum/action/cached_actions = other_actions?.Copy()
```

don't forget the other references as well.

**Review by psykzz — APPROVED** [review]

**Review by psykzz — CHANGES_REQUESTED** [review]
Agreed to move the relevant abilities to should_show instead of removing / adding them each time.

File: code/datums/actions/xeno_action.dm [hunk 2]
```diff
@@ -34,11 +34,13 @@
...
 
-/datum/action/xeno_action/remove_action(mob/living/L)
+/datum/action/xeno_action/remove_action(mob/living/L, keep_cooldown = FALSE)
 	if(keybind_signal)
 		UnregisterSignal(L, keybind_signal)
+	if(alternate_keybind_signal)
+		UnregisterSignal(L, alternate_keybind_signal)
 	UnregisterSignal(L, COMSIG_XENOMORPH_ABILITY_ON_UPGRADE)
-	if(cooldown_id)
+	if(cooldown_id && !keep_cooldown)
```
Comment (psykzz) [inline code/datums/actions/xeno_action.dm:43]: Do you still plan to keep this?

Comment (BraveMole) [reply code/datums/actions/xeno_action.dm:43]: ah no oups

**Review by psykzz — APPROVED** [review]

**Pull request merged**